qiskit/circuit/library/hidden_linear_function.py

```diff
@@ -77,22 +83,78 @@ def __init__(self, adjacency_matrix: Union[List[List[int]], np.ndarray]) -> None
         Raises:
             CircuitError: If A is not symmetric.
         """
-        adjacency_matrix = np.asarray(adjacency_matrix)
-        if not np.allclose(adjacency_matrix, adjacency_matrix.transpose()):
-            raise CircuitError("The adjacency matrix must be symmetric.")
-
-        num_qubits = len(adjacency_matrix)
-        circuit = QuantumCircuit(num_qubits, name=f"hlf: {adjacency_matrix}")
-
-        circuit.h(range(num_qubits))
-        for i in range(num_qubits):
-            for j in range(i + 1, num_qubits):
-                if adjacency_matrix[i][j]:
-                    circuit.cz(i, j)
-        for i in range(num_qubits):
-            if adjacency_matrix[i][i]:
-                circuit.s(i)
-        circuit.h(range(num_qubits))
-
+        circuit = hidden_linear_function(adjacency_matrix)
         super().__init__(*circuit.qregs, name=circuit.name)
-        self.compose(circuit.to_gate(), qubits=self.qubits, inplace=True)
+        self.compose(circuit, qubits=self.qubits, inplace=True)
```

Comment:
We probably still want this wrapping to keep `HiddenLinearFunction` unchanged
```suggestion
        self.append(circuit.to_gate(), self.qubits)
```

Reply:
Fixed.